src/peft/tuners/lora/layer.py

```diff
@@ -711,17 +717,24 @@ def forward(self, x: torch.Tensor, *args: Any, **kwargs: Any) -> torch.Tensor:
         else:
             result = self.base_layer(x, *args, **kwargs)
             torch_result_dtype = result.dtype
+
+            lora_A_keys = self.lora_A.keys()
             for active_adapter in self.active_adapters:
-                if active_adapter not in self.lora_A.keys():
+                if active_adapter not in lora_A_keys:
                     continue
```

Comment:
@BenjaminBossan  I did micro benchmarks on this code too. And lru cache and old code had exact same performance which is very strange as the `dict.keys` must make a copy/clone of at the very least the `list` for the keys. But benchmarks shows there is no `speed` difference. Still, I am a firm believer that nothing is magic so I removed the unsafe `lru_cache` with a var outsid the loop so the same values list object is not allocated in a loop.